Services/Mail/classes/class.ilMailAutoCompleteSentMailsRecipientsProvider.php

```diff
@@ -12,53 +12,57 @@ class ilMailAutoCompleteSentMailsRecipientsProvider extends ilMailAutoCompleteRe
     /**
      * @var array
      */
-    protected $users_stack = array();
+    protected array $users_stack = [];
     
     /**
      * "Current" implementation of iterator interface
      * @return  array
      */
-    public function current()
+    public function current() : array
     {
         if (is_array($this->data)) {
-            return array(
-                'login' => $this->data['login'],
+            return [
+                'login'     => $this->data['login'],
                 'firstname' => '',
-                'lastname' => ''
-            );
-        } elseif (count($this->users_stack) > 0) {
-            return array(
-                'login' => array_shift($this->users_stack),
+                'lastname'  => '',
+            ];
+        }
+
+        if (count($this->users_stack) > 0) {
+            return [
+                'login'     => array_shift($this->users_stack),
                 'firstname' => '',
-                'lastname' => ''
-            );
+                'lastname'  => '',
+            ];
         }
     }
 
     /**
      * "Key" implementation of iterator interface
      * @return string
      */
-    public function key()
+    public function key() : string
     {
         if (is_array($this->data)) {
             return $this->data['login'];
-        } elseif (count($this->users_stack) > 0) {
+        }
+
+        if (count($this->users_stack) > 0) {
             return $this->users_stack[0];
         }
     }
 
     /**
      * @return bool
      */
-    public function valid()
+    public function valid() : bool
     {
         $this->data = $this->db->fetchAssoc($this->res);
         if (
             is_array($this->data) &&
             (
-                strpos($this->data['login'], ',') !== false ||
-                strpos($this->data['login'], ';') !== false
+                str_contains($this->data['login'], ',') ||
```

Comment:
`str_contains` is not supported in PHP 7.4 (https://www.php.net/manual/en/function.str-contains.php), which we'll have to support as well in ILIAS 8.